test/integration/builder.go

```diff
@@ -0,0 +1,158 @@
+// +build integration
+
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type wdModifier func(*v1alpha2.WorkloadDefinition)
+
+func wdNameAndDef(n string) wdModifier {
+	return func(wd *v1alpha2.WorkloadDefinition) {
+		wd.ObjectMeta.Name = n
+		wd.Spec.Reference = v1alpha2.DefinitionReference{
+			Name: n,
+		}
+	}
+}
+
+func wd(m ...wdModifier) *v1alpha2.WorkloadDefinition {
+	w := &v1alpha2.WorkloadDefinition{
+		TypeMeta: v1.TypeMeta{
+			Kind:       v1alpha2.WorkloadDefinitionKind,
+			APIVersion: v1alpha2.SchemeGroupVersion.String(),
+		},
+	}
+
+	for _, fn := range m {
+		fn(w)
+	}
+	return w
+}
+
+type compModifier func(*v1alpha2.Component)
+
+func compWithName(n string) compModifier {
```

Comment:
These functions seem will be more useful in a helper library than in the test. It probably adds more code than saved given the light usage.